### DISCUSSION TITLE
Dynamic prefixes of a commit message and dynamic PR notes

### DISCUSSION THREAD

**Discussion by sychd** [renovatebot/renovate#22190] [Suggest an Idea]
**What would you like Renovate to be able to do?**
It would be nice to have an ability to add dynamic prefixes to commits and/or PR body notes.
It seems to be handy as you can provide any additional data to the PR.

In my case, I planned to add post upgrade script that will create jira issue and then leave ticket id in commit prefix or put a link to the issue in the PR notes.

Apart from that I see it useful for adding any kind of generated links for some build results, metrics etc; mention sub-teams in case of the code under their observe has been changed and similar stuff

**Comment by rarkins** [comment]
I recommend making this feature request specific to post upgrade tasks and not attempt to be generic. Considering that post upgrade tasks are done via some type of script/child process, you'd need to work out how exactly you'd pass that type of information back to Renovate. e.g. would you dump a stringified JSON object from your script? And if so, can you / will you sacrifice dumping log messages in order to do that? Or write it to a file, etc?

Once you work out how the post upgrade tasks could pass information back, using it in templates like commitMessage and prBodyNotes should not be the hard part.
